### PR TITLE
Frontend version

### PR DESCRIPTION
**Pull request by RickSanchezStoic**
Initial changes made.

### CODE REVIEW

File: ivy/functional/frontends/__init__.py [hunk 1]
```diff
@@ -3,3 +3,90 @@
...
+        f_version = latest_version[f]
+
+    for i in list(frontend.__dict__):
+        if hasattr(frontend.__dict__[i], "version_support"):
+            orig_name = version_extractor(i, f_version)
+            if orig_name:
+                frontend.__dict__[orig_name] = frontend.__dict__[i]
+
+
+version_handler(torch)
```
Comment (djl11) [inline ivy/functional/frontends/__init__.py:90]: `numpy` should be added too?

File: ivy/functional/frontends/__init__.py [hunk 2]
```diff
@@ -3,3 +3,90 @@
 from . import jax
 from . import torch
 from . import tensorflow
+import importlib
+
+latest_version = {"torch": "1.12"}
```
Comment (djl11) [inline ivy/functional/frontends/__init__.py:8]: other frameworks?

File: ivy/functional/frontends/__init__.py [hunk 3]
```diff
@@ -3,3 +3,90 @@
...
+    f = str(frontend.__name__)
+    f = f[f.index("frontends") + 10 :]
+    try:
+        f = importlib.import_module(f)
+        f_version = f.__version__
+    except Exception:
+        f_version = latest_version[f]
+
+    for i in list(frontend.__dict__):
+        if hasattr(frontend.__dict__[i], "version_support"):
```
Comment (djl11) [inline ivy/functional/frontends/__init__.py:84]: we should solve this without needing a `version_support` attribute. We can instead detect when `i` is a version-specific function based on the string pattern, and then update the function in the `__dict__` if it overlaps correctly with the current version; no need for this to be manually flagged.

**Review by djl11 — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/__init__.py (hunk 2 above)
Comment (RickSanchezStoic) [reply ivy/functional/frontends/__init__.py:8]: Yeah, had to add this decided to get your review first.

File: ivy/functional/frontends/__init__.py (hunk 1 above)
Comment (RickSanchezStoic) [reply ivy/functional/frontends/__init__.py:90]: Yep, slipped. Will add.

File: ivy/functional/frontends/__init__.py (hunk 3 above)
Comment (RickSanchezStoic) [reply ivy/functional/frontends/__init__.py:84]: Noted

File: ivy/functional/frontends/torch/creation_ops.py [hunk 4]
```diff
@@ -18,3 +18,26 @@ def full(
...
+    requires_grad=False,
+    memory_format=None
+):
+    ret = ivy.ones_like(input, dtype=dtype, device=device)
+    if requires_grad:
+        return ivy.variable(ret)
+    return ret
+
+
+def ones_like_v_0p3p0_to_0p3p1(input, out=None):
```
Comment (djl11) [inline ivy/functional/frontends/torch/creation_ops.py:41]: we will need another function to cover `<>_0p3p0_and_below`. If `0p3p0` is the first version released, or if this function also covers all earlier releases, then we should change this function name to `<>_0p3p1_and_below`

File: ivy/functional/frontends/__init__.py [hunk 5]
```diff
@@ -3,3 +3,87 @@
...
+    else:
+        i = name.index("_v_")
+        e = name.index("_and_")
+        version_start = name[i + 3 : e]
+        version_start = int(version_start.replace("p", ""))
+        if version <= version_start:
+            return name[0:i]
+
+
+def version_handler(frontend):
```
Comment (djl11) [inline ivy/functional/frontends/__init__.py:57]: this function should be given a more informative name, I would suggest `set_frontend_to_specific_version`

File: ivy/functional/frontends/torch/creation_ops.py [hunk 6]
```diff
@@ -18,3 +18,26 @@ def full(
...
+    ret = ivy.ones_like(input, out=None)
+    return ret
```
Comment (djl11) [inline ivy/functional/frontends/torch/creation_ops.py:43]: no need for this extra line, can all be done in 1 line.

File: ivy/functional/frontends/__init__.py [hunk 7]
```diff
@@ -3,3 +3,87 @@
 from . import jax
 from . import torch
 from . import tensorflow
+import importlib
+
+latest_version = {"torch": "1.12", "tensorflow":"2.9.0", "numpy":"1.23.2", "jax":"0.3.16"}
+
+
+def version_extractor(name, version):
```
Comment (djl11) [inline ivy/functional/frontends/__init__.py:11]: we should make this function name more informative, such as `fn_name_from_version_specific_fn_name`.

File: ivy/functional/frontends/__init__.py [hunk 8]
```diff
@@ -3,3 +3,87 @@
 from . import jax
 from . import torch
 from . import tensorflow
+import importlib
+
+latest_version = {"torch": "1.12", "tensorflow":"2.9.0", "numpy":"1.23.2", "jax":"0.3.16"}
```
Comment (djl11) [inline ivy/functional/frontends/__init__.py:8]: Do you think it is necessary for us to manually track the latest versions of each framework? Seems like a lot of manual maintenance effort.

**Review by djl11 — CHANGES_REQUESTED** [review]

**Pull request merged**